asciinema2gif

```diff
@@ -118,3 +118,5 @@ convert -delay 5 -loop 0 frames/*.png gif:- | gifsicle --colors=256 --delay=6 --
 cd - >/dev/null
 mv "${tempdir}/asciicast.gif" "${output_file}"
 echo ">> Generated in: ${output_file}"
+echo "cleaning up the frames before quiting"
+find "${tempdir}" -name \*.png -delete
```

Comment:
On another note, these temporary directories, since they were created with standard methods, should be automatically cleaned at system boot automatically.